### DISCUSSION TITLE
Unable to login with GH CLI

### DISCUSSION THREAD

**Discussion by nathankoerschner** [community/community#23452] [New to GitHub]
<p>I’ve been trying to login to the GH CLI and am running into trouble.</p>
<p>I’ve been getting this same error message each time I try to login:</p>
<pre><code class="lang-auto">In order to use GitHub's API, you will need to login with your GitHub account.
? Enter your GitHub user nathanabram
? Enter your GitHub password [hidden]
Error: Error creating GitHub token
Not Found
    at createNewOathToken (/Users/nathan/.nvm/versions/node/v16.15.0/lib/node_modules/gh/lib/github.js:119:15)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async getToken (/Users/nathan/.nvm/versions/node/v16.15.0/lib/node_modules/gh/lib/github.js:52:17)
    at async Object.getGitHubInstance (/Users/nathan/.nvm/versions/node/v16.15.0/lib/node_modules/gh/lib/github.js:31:15)
    at async /Users/nathan/.nvm/versions/node/v16.15.0/lib/node_modules/gh/lib/cmd.js:120:28
    at async /Users/nathan/.nvm/versions/node/v16.15.0/lib/node_modules/gh/lib/cmd.js:171:25
</code></pre>
<p>So far I’ve tried removing the github-related dotfiles from my root directory, and I’ve uninstalled and reinstalled via brew several times. I’m running this on a late 2019 Macbook Pro.</p>
<p>Any leads on what to try to get this working would be very appreciated!</p>

**Comment by mpboom** [comment]
<p>What’s your output for the following command?</p>
<pre><code class="lang-auto">$ gh --version
</code></pre>

**Reply by hozayves** [reply]
`gh 2.8.9`

**Reply by mpboom** [reply]
@hozayves this question is marked as solved and very old. If you still need help, I'd recommend asking a new question.

**Comment by nathankoerschner** [comment]
<pre><code class="lang-auto">Trey:~ nathan$ gh --version
gh 2.8.9

</code></pre>

**Comment by mpboom** [comment]
<p>That looks like an alright version of CLI. Did you make sure you’re using a <a href="https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token" rel="noopener nofollow ugc">Personal Access Token</a> and <em>not</em> your accounts’ password? Additionally, you could try authenticating via a browser (which is the default behavior when running <a href="https://cli.github.com/manual/gh_auth_login" rel="noopener nofollow ugc"><code>gh auth login</code></a>).</p>

**Reply by hozayves** [reply]
Have you an answer because I've the question????

**Reply by mpboom** [reply]
@hozayves this question is marked as solved and very old. If you still need help, I'd recommend asking a new question.

**Comment by nathankoerschner** [comment]
<p>That’s right—I did at first think it was my regular password, but then tried a Peronsal Access Token.</p>
<p>Running <code>gh auth login</code> gives the following output:</p>
<pre><code class="lang-auto">Trey:~ nathan$ gh auth login
TypeError: Cannot read properties of undefined (reading 'options')
    at getAvailableArgsOnCmd (/Users/nathan/.nvm/versions/node/v18.2.0/lib/node_modules/gh/lib/cmd.js:203:33)
    at /Users/nathan/.nvm/versions/node/v18.2.0/lib/node_modules/gh/lib/cmd.js:164:22
    at Resolved.Resolved$interpret [as _interpret] (/Users/nathan/.nvm/versions/node/v18.2.0/lib/node_modules/gh/node_modules/fluture/index.js:663:5)
    at drain (/Users/nathan/.nvm/versions/node/v18.2.0/lib/node_modules/gh/node_modules/fluture/index.js:596:23)
    at settle (/Users/nathan/.nvm/versions/node/v18.2.0/lib/node_modules/gh/node_modules/fluture/index.js:509:17)
    at resolved (/Users/nathan/.nvm/versions/node/v18.2.0/lib/node_modules/gh/node_modules/fluture/index.js:525:7)
    at EncaseN3$res (/Users/nathan/.nvm/versions/node/v18.2.0/lib/node_modules/gh/node_modules/fluture/index.js:1550:11)
    at EncaseN$done (/Users/nathan/.nvm/versions/node/v18.2.0/lib/node_modules/gh/node_modules/fluture/index.js:1553:11)
    at FSReqCallback.oncomplete (node:fs:194:23)

</code></pre>

**Comment by airtower-luna** [comment]
<p>I think the thing to look at is why you’re getting Node/nvm errors. The official <code>gh</code> is written in Go and should be a binary. Unless there’s some weird wrapper in the MacOS installation whatever you’re calling is <em>not</em> the GitHub CLI.</p>

**Comment by nathankoerschner** [comment]
<p>Ah, thank you. I deleted that file in my node bin, but when I call <code>gh</code> it is still trying to find it:</p>
<pre><code class="lang-auto">Trey:~ nathan$ gh
-bash: /Users/nathan/.nvm/versions/node/v18.2.0/bin/gh: No such file or directory
</code></pre>
<p>Do you know if there’s a place in the OS or the shell where these “routes” to files are defined? In other words, is there a place where this mistaken ‘route’ is defined that I could delete?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Your shell (Bash) seems to still have the old binary location for <code>gh</code> cached, use <code>hash -d gh</code> to make it forget.</p>
<p>The list of places that are searched in general is the <code>PATH</code>, run <code>echo $PATH</code> to see it (colon separated list). It’s usually set during login and possibly modified when you start the shell. For Bash check <code>.bashrc</code> in your home, but there may be other system-dependent locations which I don’t know for MacOS.</p>

**Comment by nathankoerschner** [comment]
<aside class="quote group-Advisory-Council" data-username="airtower-luna" data-post="8" data-topic="254169">
<div class="title">
<div class="quote-controls"></div>
 airtower-luna:</div>
<blockquote>
<p>d in general is the <code>PATH</code> , run <code>echo $PATH</code> to see it (colon separated list). It’s usually set during login and possibly modified when you start the shell. For Bash check <code>.bashrc</code> in your home, but</p>
</blockquote>
</aside>
<p>Amazing, thank you so much!</p>
<p>Unrelated to the topic at hand, but this <code>$PATH</code> stuff seems important to understand:</p>
<p>I’m seeing the colon separated list. Each of these is a path to some place. So just to make sure I’m understanding correctly, this is a list of the search heirarchy which the OS follows to find a file called <code>node</code>, <code>ruby</code>, <code>gh</code>, etc. when I call those from the shell?</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="nathanabram" data-post="9" data-topic="254169">
<div class="title">
<div class="quote-controls"></div>
 nathanabram:</div>
<blockquote>
<p>So just to make sure I’m understanding correctly, this is a list of the search heirarchy which the OS follows to find a file called <code>node</code> , <code>ruby</code> , <code>gh</code> , etc. when I call those from the shell?</p>
</blockquote>
</aside>
<p>Yes, exactly. The directories in your <code>PATH</code> are searched in order for the binary name, and the first match gets called. If you want to call another binary of the same name, you need to either call it by its full path, or adjust the <code>PATH</code>.</p>

**Comment by Aadmaa** [comment]
If this happens to you in Windows, try this on the command line: 

```
where gh
```
Now you might see something like: 

```
C:\Users\USER\AppData\Roaming\npm\gh
C:\Users\adamn\AppData\Roaming\npm\gh.cmd
```

So you can change the names of those files to get them out of the way:

```
rename C:\Users\adamn\AppData\Roaming\npm\gh.cmd gh_bk.cmd
rename C:\Users\adamn\AppData\Roaming\npm\gh gh_bk
```

You can also go to your PATH settings and move the path to GH up to the top of the PATH list. It's like:

```
C:\Program Files\GitHub CLI
```

You will need to open a fresh Command window after you update the path if you do it from Windows settings.

Then when you run "gh" it ought to be the real gh

**Reply by himanshurai24** [reply]
perfect.....worked for me in windows : )

**Comment by midhunadarvin** [comment]
I got this issue when i tried to install `gh` using `npm` .

I uninstalled and  used `brew` to install `gh`. Then it worked.
